runtime/src/message_processor.rs

```diff
@@ -58,10 +59,41 @@ fn get_subset_unchecked_mut<'a, T>(
         .collect())
 }
 
+pub struct PreAccount {
```

Comment:
The name `PreAccount` confuses me.  Without any more context, I read this as "something that becomes an account one day".

Reply:
ComparandAccount?  AccountLight?  DataOptionalAccont?  (you can tell I'm naming challenged).  Any suggestions?

Reply:
heh, what is *it*?

Reply:
Is a snapshot of the account information before the instruction is executed that may nor may not contain the account data depending on if we need to verify it after the instruction or now.

Reply:
CheckAccount?  AccountCheck?

Reply:
🤔 `AccountInternal`?  `AccountSnapshot` (I hate it, snapshot is overloaded)?  `RuntimeAccount`?  `MessageProcessorAccount`?

Reply:
VerifyAccount?

Reply:
`PreInstructionAccount`?  `AccountPreInstruction'`?

Reply:
how about:
```rust
// The relevant state of an account before an Instruction executes, used  
// to verify account integrity after the Instruction completes 
struct PreInstructionAccount {
    .
    .
}
```

Reply:
It's unusual to name a data structure after a particular use of it.

Reply:
If we didn't already have the term "snapshot", I'd call it `AccountSnapshot`.